algorithms/appo/model_utils.py

```diff
@@ -270,12 +271,20 @@ def __init__(self, cfg, obs_space, timing):
 
         if cfg.encoder_subtype == 'mlp_quads':
             fc_encoder_layer = cfg.hidden_size
-            encoder_layers = [
-                nn.Linear(obs_shape.obs[0], fc_encoder_layer),
-                nonlinearity(cfg),
-                nn.Linear(fc_encoder_layer, fc_encoder_layer),
-                nonlinearity(cfg),
-            ]
+            if cfg.use_spectral_norm:
+                encoder_layers = [
+                    spectral_norm(nn.Linear(obs_shape.obs[0], fc_encoder_layer)),
+                    nonlinearity(cfg),
+                    spectral_norm(nn.Linear(fc_encoder_layer, fc_encoder_layer)),
+                    nonlinearity(cfg),
+                ]
+            else:
+                encoder_layers = [
+                    nn.Linear(obs_shape.obs[0], fc_encoder_layer),
+                    nonlinearity(cfg),
+                    nn.Linear(fc_encoder_layer, fc_encoder_layer),
+                    nonlinearity(cfg),
+                ]
```

Comment:
This definitely works, but quite a lot of code duplication where we can easily avoid it.

How about defining a function `fc_layer` somewhere in model_utils.py which will return either wrapped or unwrapped nn.Linear depending on the config?
Then we can use it everywhere Linear layers are used (except for the outputs of the policies, I think there FC layers should be without the spectral norm)